src/doublylinkedlist.c

```diff
@@ -105,3 +112,30 @@ void DList_InsertHeadList(PDLIST_ENTRY listHead, PDLIST_ENTRY entry)
     listHead->Flink->Blink = entry;
     listHead->Flink = entry;
 }
+
+IMPLEMENT_MOCKABLE_FUNCTION(, DLIST_FIND_RESULT, DList_Find, PDLIST_ENTRY, listHead, DLIST_MATCH_FUNCTION, matchFunction, const void*, matchContext, PDLIST_ENTRY*, foundEntry)
+{
+    (void)listHead;
+    (void)matchFunction;
+    (void)matchContext;
+    (void)foundEntry;
+    return DLIST_FIND_RESULT_NOT_FOUND;
+}
+
+IMPLEMENT_MOCKABLE_FUNCTION(, int, DList_RemoveIf, PDLIST_ENTRY, listHead, DLIST_CONDITION_FUNCTION, conditionFunction, const void*, conditionContext, DLIST_ENTRY_DESTROY_FUNCTION, destroyFunction, const void*, destroyContext)
```

Comment:
>DLIST_ENTRY_DESTROY_FUNCTION, destroyFunction, [](http://example.com/codeflow?start=151&length=46)

I am thinking more and more that this parameter is not needed here.

Reply:
REmove means just that - remove. DEstroy means "deallocate used resources" and frankly, user might not want to do that.

What if user want just to move the DLIST_ENTRY from one list to another??

Reply:
Removing DList_RemoveIf in next iteration.